icechunk/src/change_set.rs

```diff
@@ -114,6 +119,30 @@ impl ChangeSet {
                 self.updated_arrays.insert(node_id.clone(), array_data);
             }
         }
+
+        // update existing splits
+        if let Some(manifests) = self.set_chunks.remove(node_id) {
+            let mut new_manifests =
+                HashMap::<ManifestExtents, SplitManifest>::with_capacity(
+                    new_splits.len(),
+                );
+            for (old_extents, chunks) in manifests.into_iter() {
+                // FIXME: these extents had better be compatible
+                // test_cases: increasing size of (multiple) dimensions
+                //             decreasing size of (multiple) dimensions
+                //
+                new_splits
```

Comment:
yeah this is ugly. Since we track extents on the change set, we need to update them for any `resize` 🤦🏾‍♂️